packages/vscode/src/binaryValidator.ts

```diff
@@ -0,0 +1,64 @@
+import { ExtensionContext, window } from 'vscode'
+
+export const PRISMA_ALLOWED_BINS_KEY = 'prismaAllowedBins'
+
+export type BinaryState = {
+  allowed: boolean
+}
+
+export enum BinaryPopupStatus {
+  Allow,
+  Deny,
+}
+
+export type BinaryStorage = { libs: { [key: string]: BinaryState } }
+
+export const checkAndAskForBinaryExecution = async (
+  context: ExtensionContext,
+  binaryPath: string | undefined,
+  bins: BinaryStorage | undefined,
+): Promise<boolean> => {
+  // empty binary path means we are using the default binary
+  if (!binaryPath || binaryPath === '') {
+    return true
+  }
+  const existingBinaryState = bins?.libs[binaryPath]
+  if (!existingBinaryState) {
+    const item = await window.showInformationMessage(
+      `Your VS Code workspace is defining a custom binary path for the Prisma extension: '${binaryPath}'. Do you trust this binary and allow the extension to execute it?`,
+      {
+        modal: true,
+      },
+      { title: 'Allow', value: BinaryPopupStatus.Allow },
+      { title: 'Deny', value: BinaryPopupStatus.Deny },
+    )
+    if (item === undefined) {
+      // dialog was cancelled
+      return false
+    }
+
+    let result = false
+
+    if (item.value === BinaryPopupStatus.Allow) {
+      result = true
+    }
+
+    if (item.value === BinaryPopupStatus.Deny) {
```

Comment:
This if-block can be removed I think, you are already setting result to false, so you don't need to check for BinaryPopupStatus.Deny here to set result to false again.

Reply:
Oh yeah, I initially added more energy variants so this check remained. Removed!